furryfeet_be/furryfeet/views.py

```diff
@@ -436,4 +437,222 @@ def post(self, request):
+
+        # get the appointment date from request data
+        date_str = request.data.get('date')
+        try:
+            date = datetime.strptime(date_str, iso_format)
+        except ValueError:
+            return Response({'error': 'Invalid date format. Please use ISO format.'}, status=status.HTTP_400_BAD_REQUEST)
+
+        # create the appointment object
+        appointment = Appointment.objects.create(
+            dog_walker=dog_walker,
+            dog=dog,
+            date=date,
+        )
+
+        # serialize the appointment object and return the response
+        serializer = AppointmentSerializer(appointment)
+        return Response(serializer.data, status=status.HTTP_201_CREATED)
+
+
+
+
+class AppointmentListView(APIView):
+    def get(self, request):
+        token = request.COOKIES.get("jwt")
+
+        if not token:
+            raise AuthenticationFailed("Unauthenticated!")
+
+        try:
+            payload = jwt.decode(token, "secret", algorithms=["HS256"])
+        except jwt.ExpiredSignatureError:
+            raise AuthenticationFailed("Unauthenticated!")
+
+        user = User.objects.get(id=payload["id"])
+
+        if not user.is_dog_walker:
+            return Response({"error": "Current user is not a dog walker."}, status=status.HTTP_403_FORBIDDEN)
+
+        appointments = Appointment.objects.filter(dog_walker=user).all()
+        serializer = AppointmentSerializer(appointments, many=True)
+
+        return Response(serializer.data, status=status.HTTP_200_OK)
+
+
+class AppointmentDetailView(APIView):
+    def get(self, request, appointment_id):
+        token = request.COOKIES.get('jwt')
+
+        if not token:
+            raise AuthenticationFailed('Unauthenticated!')
+
+        try:
+            payload = jwt.decode(token, 'secret', algorithms=['HS256'])
+        except jwt.ExpiredSignatureError:
+            raise AuthenticationFailed('Unauthenticated!')
+
+        try:
+            appointment = Appointment.objects.get(id=appointment_id)
+        except Appointment.DoesNotExist:
+            return Response({'error': 'Invalid appointment ID.'}, status=status.HTTP_400_BAD_REQUEST)
+
+        if appointment.dog_walker.id != payload['id']:
+            return Response({'error': 'Unauthorized to view this appointment.'}, status=status.HTTP_403_FORBIDDEN)
+
+        serializer = AppointmentSerializer(appointment)
+        return Response(serializer.data, status=status.HTTP_200_OK)
+
+
+class AppointmentDeleteView(APIView):
+    def delete(self, request, appointment_id):
+        token = request.COOKIES.get('jwt')
+        if not token:
+            raise AuthenticationFailed('Unauthenticated')
+
+        try:
+            payload = jwt.decode(token, 'secret', algorithms=['HS256'])
+        except jwt.ExpiredSignatureError:
+            raise AuthenticationFailed('Unauthenticated')
+
+        appointment = get_object_or_404(Appointment, pk=appointment_id)
```

Comment:
try being consistent with what you write. you dont have to change this here but for future reference either use try except or the django shortcut so that the code makes sense